eb_hooks.py

```diff
@@ -516,6 +520,20 @@ def post_module_hook_zen4_gcccore1220(self, *args, **kwargs):
                 del self.initial_environ[EESSI_IGNORE_ZEN4_GCC1220_ENVVAR]
 
 
+def post_easyblock_hook_copy_easybuild_subdir(self, *args, **kwargs):
```

Comment:
Should this include a version check, i.e. do we only run the hook for EESSI 2025.06 (and newer)?

It doesn't make a lot of sense to only do this for new installations in 2023.06 (we still won't have the information for the current installations, which defeats the point of having this central `reprod` dir), and 2023.06 has older EB versions that won't support this hook (though our policy states that we should always use the most recent version).

Reply:
It would be nicer if we could make the hook optional somehow, i.e. just skip it for 2023.06 / older EB versions (and perhaps print a warning). But I'm not sure if that can be done, as EB itself will fail when that function name is defined. @ocaisa you know more about the framework, do you know if that's possible somehow? Or do you have another suggestion to prevent that we break all builds with older EB versions?

Reply:
In this specific case, I think basically the hook will just never get called (it literally doesn't exist in framework until 5.1.1). That's probably worth a comment above the `post_easyblock_hook`.

Reply:
But apparently there is some check, as it really fails with older versions, see https://github.com/EESSI/software-layer-scripts/pull/24#discussion_r2192384809.